### ISSUE TITLE
Possible discrepancy in FocalLoss calculation

### ISSUE THREAD

**Issue by RPrudden** [yhenon/pytorch-retinanet#246]
Hello. Many thanks for this useful work!

I have found a possible issue in the calculation of the FocalLoss. In the code below, 

```# compute the loss for classification
            targets = torch.ones(classification.shape) * -1
            targets = targets.cuda()

            targets[torch.lt(IoU_max, 0.4), :] = 0

            positive_indices = torch.ge(IoU_max, 0.5)

            num_positive_anchors = positive_indices.sum()

            assigned_annotations = bbox_annotation[IoU_argmax, :]

            targets[positive_indices, :] = 0
            targets[positive_indices, assigned_annotations[positive_indices, 4].long()] = 1

            ...

            bce = -(targets * torch.log(classification) + (1.0 - targets) * torch.log(1.0 - classification))
```

It seems as though `targets` will contain a mixture of `1`, `0` and `-1` values, but the targets for BCE loss should always be between 0 and 1. It would seem to me that the thresholds should match, e.g.

```
            targets[torch.lt(IoU_max, 0.5), :] = 0

            positive_indices = torch.ge(IoU_max, 0.5)
```

I would be grateful to know if there is something I have misunderstood.

**Comment by mjun0812** [comment]
Your code is smart, but the original code is also correct.

Please look at these lines.

https://github.com/yhenon/pytorch-retinanet/blob/0348a9d57b279e3b5b235461b472d37da5feec3d/retinanet/losses.py#L120-L123

These lines means setting `target[target == -1] = 0`.

**Comment by RPrudden** [comment]
You're quite right, thanks for the response!
